### PR TITLE
#1393 Scale up Sensor column on resize &  Log file rotation method

### PR DESCRIPTION
**Pull request by hcabel**
Originate from issue #1393 

# Scale up
Due to the hierarchy and the length of the sensors name.
I've changed so that **when the TreeView is being resized the new available space goes to the first column** _(aka "Sensor")_
Before it was scaling up the last _("Max")_ column.
I've also swap how column where resized, because it you could overflow the total tree view width, and the last column was hard to resize _(due to the separator being either overflowed or on the edge)_.
So now the column separators a resizing the column to the right instead of the left. (feel much more natural)
Except for the first column which is resized by the total width of the tree view as mentioned above.

# Log file rotation
Before the CSV file was rotated daily, and if a new log session was being started the same day it would had append the new data to the end of this file.
This is a bit annoying in certain situation, and required the user to always make sure the last file is deleted.
So I've added an extra option to the logger that allow the user to specify the desired behaviour
#####  Daily
Rotate every day, just like before
##### Per session _(default)_
Rotate each time a new logging session is started.

### CODE REVIEW

File: LibreHardwareMonitor/UI/MainForm.Designer.cs [hunk 1]
```diff
@@ -446,142 +453,171 @@ private void InitializeComponent()
...
-            // 
+            //
             // logSensorsMenuItem
-            // 
+            //
             this.logSensorsMenuItem.Name = "logSensorsMenuItem";
             this.logSensorsMenuItem.Size = new System.Drawing.Size(221, 22);
             this.logSensorsMenuItem.Text = "Log Sensors";
-            // 
+			//
```
Comment (PhyxionNL) [inline LibreHardwareMonitor/UI/MainForm.Designer.cs:590]: Also not formatted right/

File: LibreHardwareMonitor/UI/MainForm.Designer.cs [hunk 2]
```diff
@@ -446,142 +453,171 @@ private void InitializeComponent()
...
+			this.perSessionFileRotationMenuItem.ToolTipText = "Create a new log file for each logging session";
+			this.perSessionFileRotationMenuItem.Click += new System.EventHandler(this.perSessionFileRotationMenuItem_Click);
+			//
+			// dailyFileRotationMenuItem
+			//
+			this.dailyFileRotationMenuItem.CheckOnClick = true;
+			this.dailyFileRotationMenuItem.Name = "dailyFileRotationMenuItem";
+			this.dailyFileRotationMenuItem.Size = new System.Drawing.Size(198, 24);
+			this.dailyFileRotationMenuItem.Text = "Daily";
+			this.dailyFileRotationMenuItem.ToolTipText = "Create a new log file every day.\n Note: If a file already exists for the current day, the new data will be appended at the end.";
```
Comment (PhyxionNL) [inline LibreHardwareMonitor/UI/MainForm.Designer.cs:616]:
```suggestion
			this.dailyFileRotationMenuItem.ToolTipText = "Create a new log file every day.\n If a file already exists for the current day, new data will be appended at the end.";
```

File: LibreHardwareMonitor/Utilities/Logger.cs [hunk 3]
```diff
@@ -170,13 +181,30 @@ public void Log()
...
-            _day = now.Date;
-            _fileName = GetFileName(_day);
-
-            if (!OpenExistingLogFile())
-                CreateNewLogFile();
+            case LoggerFileRotationMethod.PerSession:
+                // Create file if it does not exist or the logging interval has passed (+ some margin)
+                if (!File.Exists(_fileName) || now - _lastLoggedTime > (LoggingInterval + TimeSpan.FromMilliseconds(100)))
+                {
+                    uint sessionNumber = 0;
```
Comment (PhyxionNL) [inline LibreHardwareMonitor/Utilities/Logger.cs:190]: This logic is not right, you're now also having the first session the same as the day one. The first session number should start at 1.

**Review by PhyxionNL — CHANGES_REQUESTED** [review]

File: LibreHardwareMonitor/Utilities/Logger.cs (hunk 3 above)
Comment (hcabel) [reply LibreHardwareMonitor/Utilities/Logger.cs:190]: Good catch

File: LibreHardwareMonitor/Utilities/Logger.cs [hunk 4]
```diff
@@ -170,13 +181,30 @@ public void Log()
         if (_lastLoggedTime + LoggingInterval - new TimeSpan(5000000) > now)
             return;
 
-        if (_day != now.Date || !File.Exists(_fileName))
+        switch (FileRotationMethod)
```
[the remark this answers is not in the document]
Comment (hcabel) [reply LibreHardwareMonitor/Utilities/Logger.cs:184]: I'm not sure what you meant by "reformat".
For the other there was some tab indentation, that I've fixed.
But for this one there is none (for some reason) so I'm not sure what you want me to do ? `case` aligned with `switch` ?
Please let me know and I'll fix it :)

**Comment by hcabel** [comment]
Everything should be good now (except maybe Logger.cs, let me know)
My IDE had whitespace diff turned off by default, I've just turned it ON and now Ican see the mess my auto formatting made (even though I've tried to only commit what was necessary ^^) sry about that

**Comment by iNgeon** [comment]
Is this related to the struggles on v0.9.4 ? https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/issues/154#issuecomment-2551613486

**Pull request merged**